core/controller/src/main/scala/whisk/core/loadBalancer/ShardingContainerPoolBalancer.scala

```diff
@@ -137,7 +137,18 @@ import scala.util.{Failure, Success}
  * - If a loadbalancer leaves or joins the cluster, all state is removed and created from scratch. Those events should
  *   not happen often.
  */
-class ShardingContainerPoolBalancer(config: WhiskConfig, controllerInstance: ControllerInstanceId)(
+class ShardingContainerPoolBalancer(
+  config: WhiskConfig,
+  controllerInstance: ControllerInstanceId,
+  private val feedFactory: (ActorRefFactory, MessagingProvider, (Array[Byte]) => Future[Unit]) => ActorRef,
```

Comment:
@tysonnorris May be we can use traits here instead of function arguments. It makes it easy in IDE to see what is the implementation and move back and forth easily